zenpy/lib/generator.py

```diff
@@ -200,6 +201,19 @@ def process_page(self):
             search_results.append(self.response_handler.api._object_mapping.object_from_json(object_type, object_json))
         return search_results
 
+    def get_next_page(self, page_num, page_size):
+        """ Retrieve the next page of results. """
+        url = self._response_json.get(self.next_page_attr, None)
+        if url is None:
+            raise StopIteration()
+        params, url = self.process_url(page_num, page_size, url)
+        try:
+            response = self.response_handler.api._get(url, raw_response=True, params=params)
+        except SearchResponseLimitExceeded:
```

Comment:
Instead if copy pasting this method you could do something like:

```python
def get_next_page(self, page_num, page_size):
    try:
        return super().get_next_page(self, page_num, page_size)
    except SearchResponseLimitExceeded:
        log.error('This search has resulted in more results than zendesk allows. We got what we could.')
        raise StopIteration()
```

Reply:
oh. duh. rookie mistake

Reply:
Would you like py2 or py3 syntax for `super()`? I did it as py2 because it looks like you're still testing against it in travis, but I'm happy to change it

Reply:
Yep py2 is syntax is fine